helpers/plugins.js

```diff
@@ -0,0 +1,26 @@
+const pluginSets = new Map();
+
+export function getPlugin(setKey, pluginKey) {
```

Comment:
In hindsight, `getPlugins` could have been implemented to accept a plugin key. In such a case, it would still return an array, but it would only contain plugins within the set that have a certain plugin key. This would effectively mean that different plugins could legit have the same key, which is _interesting_. We have no use case for this, and we could always add it later.

Reply:
Yeah... wondering if for now we should be throwing if two plugins get registered with the same key? It was likely a mistake and would cause a lot of confusion as to why it's not working... and even worse someone could accidentally overwrite someone else's plugin without knowing it.

Reply:
Yeah, that's a good idea. If a some functionality requires something akin to plugins with the same key, they can just not use the `key` and they can define something on their plugin API.

Reply:
Added this along with a couple tests for it.